### PR TITLE
DDF-6049 Move SecureWebSocketServlet into DDF

### PR DESCRIPTION
**Pull request by bakejeyner**
[DDF-UI Pull Request](https://github.com/codice/ddf-ui/pull/219)

#### What does this PR do?
Moves the `SecureWebSocketServlet` into DDF and changes the way the web sockets checks to see if a user is logged in. The web sockets now store a reference to the http session (which contains the login information), and when that session is invalid it means the user has logged out.

#### Who is reviewing it? 
@SmithJosh 

#### Ask 2 committers to review/merge the PR and tag them here.

@blen-desta 
@garrettfreibott 
@stustison 

#### How should this be tested?
Feel free to contact me with any questions.

Setup:
1. Install the standard profile.
1. Build and install [this DDF-UI](https://github.com/codice/ddf-ui/pull/219). **Make sure that ddf.version in the DDF-UI root pom matches this version of DDF.**

Test w/ Basic:
1. Login to the admin console with basic credentials.
1. Navigate to `System -> Configuration -> Catalog UI Search` and check `Enable Web Sockets`.
1. *In a new tab* login to intrigue with basic credentials.
1. Ingest some things.
1. Execute a query and verify results.
1. In the admin console tab, logout of the admin account. The goal here is to logout of the admin account with the intrigue tab still open and ready.
1. In the intrigue tab, re-execute the query. The query should fail. (Further executions may succeed due to the user being re-signed in with cached credentials, that is expected behavior)
1. Refreshing the intrigue tab should make you re-enter credentials and querying should work as expected.

(Extra Credit) Test w/ OIDC:
1. Start up keycloak and set it up a client for DDF to OIDC.
1. Login to DDF's admin console.
1. Modify the OIDC Handler configuration to contain the proper information to connect with keycoak.
1. Modify the Web Context Policy Manager to use OIDC for Web Pages and Endpoints.
1. Logout.
1. Login to intrigue.
1. Execute a query and verify results.
1. *In a new tab* login to a web page (i.e. the admin console or intrigue) and logout. The goal here is to logout of the admin account with the intrigue tab still open and ready.
1. In the intrigue tab, re-execute the query. The query should fail. Further executions of the query should also fail.
1. Refreshing the intrigue tab should make you re-enter credentials and querying should work as expected.

#### What are the relevant tickets?
Fixes: #6049

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bakejeyner** [comment]
build now

File: platform/security/servlet/security-servlet-web-socket/src/main/java/org/codice/ddf/security/servlet/web/socket/SecureWebSocketServlet.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  @Override
+  public void destroy() {
+    executor.shutdown();
+  }
+
+  /*
+   Pass the TokenHolder into the WebSocket, in order to know when the user has logged out. Can't
+   pass the Session because Jetty won't let anything check session attributes unless there's a
+   request (excluding WebSocket requests) being actively fulfilled for that session.
+  */
```
Comment (SmithJosh) [inline platform/security/servlet/security-servlet-web-socket/src/main/java/org/codice/ddf/security/servlet/web/socket/SecureWebSocketServlet.java:54]: Looks like this comment no longer applies

File: features/security/src/main/feature/feature.xml [hunk 2]
```diff
@@ -528,6 +528,13 @@ org.ops4j.pax.web.default.realmname=DDF
         <bundle>mvn:ddf.security.servlet/security-servlet-whoami/${project.version}</bundle>
     </feature>
 
+    <feature name="security-servlet-web-socket" version="${project.version}"
+             description="Defines a secure web socket servlet type.">
+        <feature>security-core-impl</feature>
```
Comment (stustison) [inline features/security/src/main/feature/feature.xml:533]: This might need to be changed now. Try rebasing and see if this actually needs impl.

**Comment by blen-desta** [comment]
Hero'd successfully ✅ (including the extra credit 😆)

**Comment by bakejeyner** [comment]
build now

File: features/security/src/main/feature/feature.xml (hunk 2 above)
Comment (bakejeyner) [reply features/security/src/main/feature/feature.xml:533]: 👍 I'll see if CI fails on it first. Scared of building atm

**Comment by bakejeyner** [comment]
CI logs seemed a little wierd; going to try re-building
build now

File: features/security/src/main/feature/feature.xml (hunk 2 above)
Comment (bakejeyner) [reply features/security/src/main/feature/feature.xml]: Manually verified it is now using artifacts from `security-core-api`. Making the change.

**Comment by bakejeyner** [comment]
build now

**Comment by bakejeyner** [comment]
build now
(formatting)

**Pull request merged**